src/tests/resistance/fuzzing_flood.rs

```diff
@@ -0,0 +1,640 @@
+use comfy_table::{modifiers::UTF8_ROUND_CORNERS, presets::UTF8_FULL, Cell, CellAlignment, Table};
+use histogram::Histogram;
+use rand::{prelude::SliceRandom, Rng};
+use rand_chacha::ChaCha8Rng;
+use std::{
+    convert::TryFrom,
+    io::{Read, Write},
+    net::SocketAddr,
+};
+use tokio::{sync::mpsc::Sender, time::Duration};
+
+use crate::{
+    helpers::{initiate_handshake, is_rejection_error, is_termination_error},
+    protocol::{
+        message::{filter::MessageFilter, Message},
+        payload::{
+            block::{Block, Headers, LocatorHashes},
+            codec::Codec,
+            Hash, Inv, Nonce,
+        },
+    },
+    setup::{
+        config::new_local_addr,
+        node::{Action, Node},
+    },
+    tests::resistance::{
+        default_fuzz_messages,
+        fuzzing_corrupted_messages::slightly_corrupted_messages,
+        fuzzing_incorrect_checksum::encode_messages_and_corrupt_checksum,
+        fuzzing_incorrect_length::encode_messages_and_corrupt_body_length_field,
+        fuzzing_random_bytes::{metadata_compliant_random_bytes, random_bytes},
+        fuzzing_zeroes::zeroes,
+        seeded_rng, COMMANDS_WITH_PAYLOADS,
+    },
+};
+
+enum Event {
+    HandshakeEstablished(Duration),
+    HandshakeRejected,
+    HandshakeError(std::io::Error),
+    ValidReply(Duration),
+    BadReply(Box<Message>, Box<Message>),
+    WriteError(std::io::Error),
+    ReadError(std::io::Error),
+    Dropped,
+    Terminated,
+    IgnoredCorrupt(Vec<u8>),
+    RejectedCorrupt,
+    RepliedToCorrupt(Box<Message>),
+    Complete,
+}
+
+#[derive(Default, Debug)]
+struct Stats {
+    handshake_accepted: u16,
+    current_connections: u16,
+    max_active_connections: u16,
+    handshake_latencies: Vec<Duration>,
+    handshake_rejected: u16,
+    peers_dropped: u16,
+    reply_latencies: Vec<Duration>,
+    corrupt_terminated: u16,
+    corrupt_rejected: u16,
+    reply_instead_of_termination: u16,
+    corrupt_ignored: u16,
+
+    reply_errors: u16,
+    read_errors: u16,
+    write_errors: u16,
+    handshake_errors: u16,
+    dangling: u16,
+}
+
+#[tokio::test(flavor = "multi_thread", worker_threads = 8)]
+async fn rising_fuzz() {
+    // ZG-RESISTANCE-006
+    //
+    // Simulate high load situations, while also sending corrupt messages.
+    //
+    // We simulate active peers which send 100 requests (and validate the replies), and finally send a corrupt message
+    // which ends the connection. We measure the latency of the replies and the type of response to the corrupt message.
+    //
+    //  *NOTE* run with `cargo test --release tests::resistance::fuzzing_flood::rising_fuzz -- --nocapture`
+    //
+    // Currently only works for zcashd as it requires block seeding.
```

Comment:
Have you tested the new zebra seeding with this yet?

Reply:
No, I didn't want to cross the two PR's - I can certainly try that though; would be interesting.